### PR TITLE
CXF-98764: Add Sanity FCR to VD connections

### PR DESCRIPTION
**Pull request by d-bhola**
-Update workflow to add secrets in GHA
-Add FCR to VD test in prod sanity suite 
-Add FCR to VD test in UAT sanity suite

### CODE REVIEW

File: .github/workflows/terratests-prod-suite-all.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ jobs:
       TEST_DATA_PROD_VIRTUAL_DEVICE_2_WAN_CONNECTION: ${{ secrets.TEST_DATA_PROD_VIRTUAL_DEVICE_2_WAN_CONNECTION }}
       TEST_DATA_PROD_VIRTUAL_DEVICE_2_AZURE_CONNECTION: ${{secrets.TEST_DATA_PROD_VIRTUAL_DEVICE_2_AZURE_CONNECTION}}
       TEST_DATA_PROD_VIRTUAL_DEVICE_2_PORT_CONNECTION: ${{secrets.TEST_DATA_PROD_VIRTUAL_DEVICE_2_PORT_CONNECTION}}
+      TEST_DATA_PROD_CLOUD_ROUTER_2_VIRTUAL_DEVICE_CONNECTION: ${{secrets.TEST_DATA_PROD_CLOUD_ROUTER_2_VIRTUAL_DEVICE_CONNECTION}}
```
Comment (thogarty) [inline .github/workflows/terratests-prod-suite-all.yaml:32]: Keep the test names next to their neighbors. Same with the `jq and echo` statements as well. That way it's clear to see the list with a quick glance without spending more time.

**Review by thogarty — CHANGES_REQUESTED** [review]
Requested change.

File: .github/workflows/terratests-prod-suite-all.yaml (hunk 1 above)
Comment (d-bhola) [reply .github/workflows/terratests-prod-suite-all.yaml]: Addressed with [this](https://github.com/equinix/terraform-equinix-fabric/pull/111/commits/2b8d064506143b64d097a00e3a7e8d2d272dd140) commit

**Comment by d-bhola** [comment]
UAT sanity is passing. Prod is failing due to the following error.   (Linked this ticket to the existing [bug ticket](https://equinixjira.atlassian.net/browse/CXF-100696) for Dragons)
```

 ---[ REQUEST ]---------------------------------------
POST /fabric/v4/connections HTTP/1.1
Host: api.equinix.com
User-Agent: HashiCorp Terraform/1.8.5 (+https://www.terraform.io) Terraform Plugin SDK/v2.34.0 terraform-provider-equinix/2.3.2 equinix-sdk-go/0.42.0
Content-Length: 504
Accept: application/json
Content-Type: application/json
X-Correlation-Id: 3CBRcn&$MSZJd90hxGwR9r#cO
X-Source: API
Accept-Encoding: gzip
                                2024-08-19T14:33:55.875-0700 [DEBUG] provider.terraform-provider-equinix_v2.3.2
{
 "aSide": {
  "accessPoint": {
   "router": {
    "uuid": "1f764d24-b400-4ef6-b3ba-3787bb5da7a7"
   },
   "type": "CLOUD_ROUTER"
  }
 },
 "bandwidth": 50,
 "name": "fcr_2_vd",
 "notifications": [
  {
   "emails": [
    "example@equinix.com",
    "test1@equinix.com"
   ],
   "type": "ALL"
  }
 ],
 "order": {
  "purchaseOrderNumber": "1-323292"
 },
 "redundancy": {
  "priority": "PRIMARY"
 },
 "type": "IP_VC",
 "zSide": {
  "accessPoint": {
   "interface": {
    "id": 5,
    "type": "NETWORK",
    "uuid": ""
   },
   "type": "VD",
   "virtualDevice": {
    "href": "",
    "name": "",
    "type": "EDGE",
    "uuid": "37cf3ffb-d436-4163-97c5-8f3607b4334a"
   }
  }
 }
}
2024-08-19T14:33:55.875-0700 [DEBUG] provider.terraform-provider-equinix_v2.3.2
-----------------------------------------------------
2024/08/19 14:33:58 [DEBUG] Equinix API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 500 Internal Server Error
Connection: close
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Content-Type: application/json
Correlationid: YT1ElvNNg6NHB2gK
Date: Mon, 19 Aug 2024 21:33:58 GMT
Expires: 0
Pragma: no-cache
Server: istio-envoy
Vary: Accept-Encoding
X-Apigee-Dp-Color: 174
X-Apigee-Environment: extprod
X-Apigee-Message-Id: a0343787-1d27-497d-b4c1-abde019e121d
X-Apigee-Organization: eqx-apg-prod
X-Apigee-Proxy: /organizations/eqx-apg-prod/environments/extprod/apiproxies/ecxv4/revisions/9
X-Apigee-Proxy-Basepath: /fabric/v4
X-Apigee-Region: us-west1
X-Apigee-Target-Latency: 1696
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 1706
X-Frame-Options: DENY
X-Request-Id: a0343787-1d27-497d-b4c1-abde019e121d
X-Xss-Protection: 1; mode=block
                                2024-08-19T14:33:58.207-0700 [DEBUG] provider.terraform-provider-equinix_v2.3.2
[
 {
  "errorCode": "EQ-4014404",
  "errorMessage": "Get Port failed",
  "help": "https://<domain.name>/help/error/EQ-4014404",
  "details": "Please open a support ticket via the ? on the bottom of every page for further assistance.",
  "additionalInfo": [
   {}
  ]
 }
]

```

File: tests/uat/uat_sanity_suite_test.go [hunk 2]
```diff
@@ -201,3 +201,17 @@ func TestVirtualDevice2PortCreateConnection_PNFV(t *testing.T) {
 	output := terraform.Output(t, terraformOptions, "port_connection_id")
 	assert.NotNil(t, output)
 }
+
+func TestCloudRouter2VirtualDeviceCreateConnection_PFCR(t *testing.T) {
```
Comment (thogarty) [inline tests/uat/uat_sanity_suite_test.go:205]: You've added this test to the PNFV Test user in GHA, but here the user is specified as PFCR. Which one is the correct one?

**Review by thogarty — COMMENTED** [review]
@d-bhola , just some confusion about UAT user. Rest looks good.

File: tests/uat/uat_sanity_suite_test.go (hunk 2 above)
Comment (d-bhola) [reply tests/uat/uat_sanity_suite_test.go:205]: Addressed with [this](https://github.com/equinix/terraform-equinix-fabric/pull/111/commits/82795304554522209684b3df70ee93fcfbf1979e) commit

**Review by thogarty — APPROVED** [review]
Approving and merging.

**Pull request merged**